### PR TITLE
neomutt: add `smime_keys` optional

### PR DESCRIPTION
**Pull request by cafkafk**
## Description of changes

This enables `smime_keys` from the neomutt source repository (found under
`/contrib`, not to be confused with the neomutt-contrib repository).

Unlike mutt, which adds `smime_keys` to the build output `/bin` by default,
neomutt currently doesn't even provide an option to expose it, and that is even
though we default to adding `man smime_keys_neomutt`!

I've set this optional to default to `true`, but that can be discussed ofc!

My reasoning was:
- It's prominently displayed on https://neomutt.org/man
- We alrady default to shipping its man pages, it would be confusing (and was
to me) confusing why I didn't have the binary in path.

I've also formatted the source file in a separate preceeding commit, I know
this can be optional, but I do feel it made optionals a lot more readable... so
I'd really prefer to keep it this way >_>

Commits:
- **neomutt: apply nixfmt-rfc-style**
- **neomutt: add `smime_keys` optional**

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by cafkafk** [comment]
The formatting change probably makes this diff really annoying to review, you can see the actual changes that matter here e37215d0b82a36925778b66bcb0f0e53fa3c3813

File: pkgs/applications/networking/mailreaders/neomutt/default.nix [hunk 1]
```diff
@@ -49,31 +95,38 @@ stdenv.mkDerivation (finalAttrs: {
...
+    ++ lib.optional enableMixmaster "--mixmaster";
+
+  postInstall =
+    ''
+      wrapProgram "$out/bin/neomutt" --prefix PATH : "$out/libexec/neomutt"
+    ''
+    + lib.optionalString enableSmimeKeys ''
+      install -m 755 $src/contrib/smime_keys $out/bin;
+      substituteInPlace $out/bin/smime_keys \
+        --replace '/usr/bin/openssl' '${openssl}/bin/openssl';
```
Comment (drupol) [inline pkgs/applications/networking/mailreaders/neomutt/default.nix:125]:
```suggestion
        --replace-fail '/usr/bin/openssl' '${openssl}/bin/openssl';
```

**Review by drupol — CHANGES_REQUESTED** [review]

**Comment by drupol** [comment]
Diff LGTM, feel free to merge, but it would be nice to replace `--replace` with `--replace-fail`.

File: pkgs/applications/networking/mailreaders/neomutt/default.nix (hunk 1 above)
Comment (cafkafk) [reply pkgs/applications/networking/mailreaders/neomutt/default.nix]: Addressed in f9b42e3, good idea

**Comment by cafkafk** [comment]
(I can't self merge when there is a change-request still hanging btw :p)

**Review by drupol — APPROVED** [review]

**Pull request merged**